pinecone/client_test.go

```diff
@@ -159,13 +185,133 @@ func (ts *ClientTests) TestAuthorizationHeaderOverridesApiKey() {
 	require.NoError(ts.T(), err)
 	require.NotNil(ts.T(), mockTransport.Req, "Expected request to be made")
 
-	apiKeyHeaderValue := mockTransport.Req.Header.Get("Api-Key")
-	authHeaderValue := mockTransport.Req.Header.Get("Authorization")
-	if authHeaderValue != "bearer fooo" {
-		ts.FailNow(fmt.Sprintf("Expected request to have header value 'bearer fooo', but got '%s'", authHeaderValue))
+	testHeaderValue := mockTransport.Req.Header.Get("test-header")
+	assert.Equal(ts.T(), "param-header", testHeaderValue, "Expected request to have header value 'param-header', but got '%s'", testHeaderValue)
+
+	os.Unsetenv("PINECONE_ADDITIONAL_HEADERS")
+}
+
+// func (ts *ClientTests) TestAuthorizationHeaderOverridesApiKey() {
```

Comment:
meant to comment out?

Reply:
Meant to remove entirely, the PR wasn't really in a finished state. 😅 The client no longer does any explicit juggling of `ApiKey` and any auth header you may have passed.